hamilflow/models/harmonic_oscillator.py

```diff
@@ -1,36 +1,29 @@
+import math
 from abc import ABC, abstractmethod
 from functools import cached_property
 from typing import Literal, Mapping, Sequence
 
 import numpy as np
 import pandas as pd
 from numpy.typing import ArrayLike
-from pydantic import BaseModel, Field, computed_field, field_validator, model_validator
-
-try:
-    from typing import Self
-except ImportError:
-    from typing_extensions import Self
+from pydantic import BaseModel, Field, computed_field, field_validator
 
 
 class HarmonicOscillatorSystem(BaseModel):
     """The params for the harmonic oscillator
 
     :cvar omega: angular frequency of the harmonic oscillator
     :cvar zeta: damping ratio
-    :cvar real: use real solution (only supported for the undamped case)
     """
 
-    omega: float
-    zeta: float = 0.0
-
-    real: bool = Field(default=True)
+    omega: float = Field()
+    zeta: float = Field(default=0.0)
 
     @computed_field  # type: ignore[misc]
     @cached_property
     def period(self) -> float:
         """period of the oscillator"""
-        return 2 * np.pi / self.omega
+        return 2 * math.pi / self.omega
```

Comment:
Is there any reason we switch to `math`? I thought there is no difference between `np.pi` and `math.pi`. 
`math` is only used here. It feels strange to important `math` just for this use case when `numpy` is already imported in this module.

Reply:
`math` is native python. Since we are using `float` instead of `np.float64`, etc, I'd like to use native python as much as possible.

Reply:
> `math` is native python. Since we are using `float` instead of `np.float64`, etc, I'd like to use native python as much as possible.

Actually, my point is, those are literally predefined constants written in the definitions and both of them are just `float`. 
Native or not, they are just explicit float number predefined in the code. 

Is `np.pi` np.float64? For me it is just float (python 3.10, numpy 1.24).

Reply:
`np.pi` is also `float` for me, not `np.float64`. However this is not fixed in the numpy docs. Even if it is, things may change in the future.

Nevertheless, this is not too important for me, I'm okay to use `np.pi` as well.

Reply:
Tried numpy 2, got the same results.

![Screenshot 2024-07-27 at 11 43 31](https://github.com/user-attachments/assets/c3a81c4c-c14a-419f-8eb5-1e2d2ddd657a)

Reply:
https://github.com/kausalflow/hamilflow/pull/64/commits/2718ef8520bd7000f438bc888547f69565d66635